core/src/main/java/com/graphhopper/routing/profiles/StringEncodedValue.java

```diff
@@ -0,0 +1,46 @@
+package com.graphhopper.routing.profiles;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * This class holds a string array and stores only a number (typically the array index via IntEncodedValue)
+ * to restore this information.
+ */
+public final class StringEncodedValue extends IntEncodedValue {
+    private final String[] map;
+
+    public StringEncodedValue(String name, List<String> values, String defaultValue) {
+        super(name, (int) Long.highestOneBit(values.size()));
+
+        // we want to use binarySearch so we need to sort the list
+        // TODO should we simply use a separate Map<String, Int>?
```

Comment:
I would think Map might be faster for larger collections and maybe even easier to use?